data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.architectcoders.aacboard.data.repository
+
+import com.architectcoders.aacboard.data.datasource.local.DeviceDataSource
+import com.architectcoders.aacboard.domain.repository.RegionRepository
+
+class RegionRepositoryImpl(
+    private val deviceDataSource: DeviceDataSource,
```

Comment:
Maybe I will use a dedicated data source for the RegionRepository as in the example.  I would use `DeviceDataSource` only for the preferences.

Reply:
Changed